libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteInterpolator.kt

```diff
@@ -31,35 +33,164 @@ internal class ReplayRouteInterpolator {
 
     /**
      * Interpolate a speed across a distance. The returned segment will include approximate
-     * once per second speed and distance calculations that can be used to create location coordinates.
+     * speed and distance calculations that can be used to create location coordinates.
+     * This will also calculate and interpolate cruising speeds, which allows for the start and
+     * end speeds to be zero.
      */
     fun interpolateSpeed(
         options: ReplayRouteOptions,
         startSpeed: Double,
         endSpeed: Double,
         distance: Double
     ): ReplayRouteSegment {
+        val segment = calculateSegmentSpeedAndDistances(options, startSpeed, endSpeed, distance)
         val speedSteps = mutableListOf<ReplayRouteStep>()
-        speedSteps.add(
-            ReplayRouteStep(
-                acceleration = 0.0,
-                speedMps = startSpeed,
-                positionMeters = 0.0
-            )
+
+        // Add the speeding up steps, if there are any
+        speedSteps.addAcceleratingSteps(
+            frequency = options.frequency,
+            startSpeed = startSpeed,
+            endSpeed = segment.maxSpeedMps,
+            distance = segment.speedUpDistance
         )
 
-        while (speedSteps.last().positionMeters < distance) {
-            estimateSpeedStep(options, speedSteps, endSpeed, distance)
-        }
+        // Add the cruising steps, if there are any
```

Comment:
Same as above re: `private fun`ction and remove unnecessary comment

Reply:
Yeah, it's already in a private function so it may already be worthless